### ISSUE TITLE
Add new data types in DataContainer

### ISSUE THREAD

**Issue by aschaeffer** [inexorgame/entity-system#55]
### New data types

In order to enable future use cases we should add these new data types in DataContainer:

- [x] vec3 (3x double, xyz, rgb)
- [x] vec4 (4x double, rgba)
- [x] byte array (textures, binary data)

### Future use cases

| Use Case | Data type(s) | Remarks / Examples |
| - | - | - |
| Entity position | vec3, vec4 | Player position, camera position, particle position, ... |
| Entity direction | vec3, vec4 | Player direction, camera direction, particle direction, ... |
| Vector arithmetic | vec3, vec4 | Velocity transformation, gravity, ... |
| Colors | vec3, vec4 | RGB or RGBA |
| Textures | byte array | Textures can be loaded from file into an attribute |
| Procedural Textures / Texture transformation | byte array | Procedural generation or transformation of textures |

### Removal

Maybe we should remove "data_float" and "data_int" for simplicity and force the use of int64 instead of int and double instead of float).

**Comment by IAmNotHanni** [comment]
For the byte array I found [this](https://stackoverflow.com/questions/16489407/how-to-create-a-byte-array-in-c) topic.

**Comment by aschaeffer** [comment]
[OpenGL Mathematics](https://glm.g-truc.net) contains a bunch of datatypes. The most important are `vec2`, `vec3` and `vec4`, but others like mat2, mat3 could be helpful, too.

```c++
#include <glm/vec2.hpp>               // vec2, bvec2, dvec2, ivec2 and uvec2
#include <glm/vec3.hpp>               // vec3, bvec3, dvec3, ivec3 and uvec3
#include <glm/vec4.hpp>               // vec4, bvec4, dvec4, ivec4 and uvec4
#include <glm/mat2x2.hpp>             // mat2, dmat2
#include <glm/mat2x3.hpp>             // mat2x3, dmat2x3
#include <glm/mat2x4.hpp>             // mat2x4, dmat2x4
#include <glm/mat3x2.hpp>             // mat3x2, dmat3x2
#include <glm/mat3x3.hpp>             // mat3, dmat3
#include <glm/mat3x4.hpp>             // mat3x4, dmat2
#include <glm/mat4x2.hpp>             // mat4x2, dmat4x2
#include <glm/mat4x3.hpp>             // mat4x3, dmat4x3
#include <glm/mat4x4.hpp>             // mat4, dmat4
```

**Comment by Fohlen** [comment]
Added basic vector types in 4e9e3d8c6d6a8d3006cfd281f5753dca8d6c5109

**Comment by Fohlen** [comment]
This will be solved by #86
